ide/lsp.client/src/org/netbeans/modules/lsp/client/bindings/ErrorProviderBridge.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.lsp.client.bindings;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import javax.swing.text.Document;
+import org.netbeans.api.lsp.Diagnostic;
+import org.netbeans.spi.editor.hints.ErrorDescription;
+import org.netbeans.spi.editor.hints.ErrorDescriptionFactory;
+import org.netbeans.spi.editor.hints.HintsController;
+import org.netbeans.spi.editor.hints.Severity;
+import org.netbeans.spi.lsp.ErrorProvider;
+import org.openide.filesystems.FileObject;
+import org.openide.util.RequestProcessor;
+import org.openide.util.WeakListeners;
+
+class ErrorProviderBridge implements Runnable, DocumentListener {
+
+    private final FileObject file;
+    private final RequestProcessor.Task task;
+    private final Collection<? extends ErrorProvider> errorProviders;
+    private final DocumentListener listener;
+
+    ErrorProviderBridge(Document doc, FileObject file, Collection<? extends ErrorProvider> errorProviders, RequestProcessor rp) {
+        this.file = file;
+        this.errorProviders = errorProviders;
+        this.task = rp.create(this);
+        this.listener = WeakListeners.create(DocumentListener.class, this, doc);
+        doc.addDocumentListener(listener);
+    }
+
+    final void start() {
+        task.schedule(0);
+    }
+
+    final void waitFinished() {
+        task.waitFinished();
+    }
+
+    @Override
+    public final void run() {
+        for (ErrorProvider p : errorProviders) {
+            if (p.getClass().getName().equals("org.netbeans.modules.editor.hints.lsp.HintsDiagnosticsProvider")) {
```

Comment:
Is it possible to use the same trick as for `JavaErrorProvider` for this one as well? Regardless of which way is used.

Reply:
I don't think that's possible. But I'll think about it and give it a try.

Reply:
Looks like it works. Done in f1678df. Thanks for the idea. Registering in VSCode's `nbcode` suite is a neat way to avoid cycles in the delegation.